### PR TITLE
feat: use thiserror for errors

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -32,193 +33,130 @@ impl<'a, T> Debug for ErrorDebug<'a, T> {
 }
 
 /// Main error type for the crate.
-#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+#[derive(Error, Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
 pub enum Error {
-    /// Access is denied for a given requester
+    /// Access denied for supplied PublicKey
+    #[error("Access Denied")]
     AccessDenied,
```
Comment (bochaco) [inline src/errors.rs]: Don't we want to provide the pk here so the consumer knows which pk was being used when the error was thrown? I know this is probably out of scope of this PR, but just thinking the direction we should take these errors to and start working on some consistency.

File: src/messaging/mod.rs [hunk 2]
```diff
@@ -640,7 +640,7 @@ mod tests {
     }
 
     #[test]
-    fn try_from() -> Result<()> {
+    fn try_from() -> anyhow::Result<()> {
```
Comment (bochaco) [inline src/messaging/mod.rs]: Perhaps simply `use anyhow::{Result, anyhow}` for the whole `test` module?

Comment (joshuef) [reply src/messaging/mod.rs]: I did try that and things got wiiild w/ compiler errors. I'll double check that's not sorted now though

Comment (joshuef) [reply src/messaging/mod.rs]: that wildness is only in the seq file. the rest ill update

**Review by bochaco — APPROVED** [review]

**Pull request merged**